src/blockdev.rs

```diff
@@ -61,17 +56,48 @@ fn get_partition_with_label(device: &str, label: &str) -> Result<Option<BlkDev>>
         eprint!("{}", String::from_utf8_lossy(&*result.stderr));
         bail!("lsblk of {} failed", device);
     }
-    let output: LsBlk =
-        serde_json::from_slice(&*result.stdout).chain_err(|| "decoding lsblk JSON")?;
+    let output = String::from_utf8(result.stdout).chain_err(|| "decoding lsblk output")?;
+
+    // walk each device in the output
+    let re = Regex::new("([A-Z-]+)=\"([^\"]+)\"").unwrap();
     let mut found: Option<BlkDev> = None;
-    for dev in output.blockdevices {
-        if dev.label.is_none() || dev.label.as_ref().unwrap() != label {
-            continue;
+    for line in output.lines() {
+        // parse key-value pairs
+        // newer versions of lsblk support JSON but CentOS 7 doesn't
+        let mut fields: HashMap<String, String> = HashMap::new();
+        for cap in re.captures_iter(line) {
+            fields.insert(cap[1].to_string(), cap[2].to_string());
```

Comment:
Should we check-and-continue for `cap.len() < 2` here?

Reply:
Neither of the match groups in the regex are optional, so we shouldn't end up here if `cap.len() < 3`.  I'm okay with the panic in that case.